R/template.R

```diff
@@ -791,5 +811,78 @@ yardstick_eval_select_dots <- function(...,
 
   out <- names(out)
 
+  set_known_selection(as_label(expr), out)
+
   out
 }
+
+# store known selections (#428) ------------------------------------------------
+is_known_selection <- function(expr_name) {
+  if (!catalog_is_available() || !in_tuning_env()) {
+    return(FALSE)
+  }
+
+  known_selections <- ns_env("tune")$tune_env$known_selections
+
+  expr_name %in% names(known_selections)
+}
+
+get_known_selection <- function(expr_name) {
+  known_selections <- ns_env("tune")$tune_env$known_selections
+
+  known_selections[[expr_name]]
+}
+
+set_known_selection <- function(expr_name, out) {
+  if (!catalog_is_available() || !in_tuning_env()) {
+    return(invisible())
+  }
+
+  tune_env <- ns_env("tune")$tune_env
+
+  if (is.null(tune_env$known_selections)) {
+    init_known_selection(tune_env)
+  }
+
+  tune_env$known_selections[[expr_name]] <- out
+
+  invisible()
+}
+
+init_known_selection <- function(tune_env) {
+  withr::defer(
+    env_bind(tune_env, known_selections = NULL),
+    envir = tune_env$progress_env
+  )
```

Comment:
See [these lines in tune](https://github.com/tidymodels/tune/blob/28a459f9f581bf35d5c28af67866588f71056b36/R/logging.R#L110) for context on this environment--`progress_env` persists while a given tune function, like `fit_resamples()` or `tune_bayes()`, runs. When that function returns, this code wipes the known selections so that we don't re-use them across different calls to tuning functions.